src/EditorFeatures/TestUtilities/Diagnostics/AbstractUserDiagnosticTest.cs

```diff
@@ -35,6 +36,16 @@ public abstract partial class AbstractUserDiagnosticTest : AbstractCodeActionOrU
         internal abstract Task<IEnumerable<Diagnostic>> GetDiagnosticsAsync(
             TestWorkspace workspace, TestParameters parameters);
 
+        protected async Task TestDiagnosticsAsync(
+            string initialMarkup, TestParameters parameters = default, params DiagnosticDescription[] expected)
+        {
+            using (var workspace = CreateWorkspaceFromOptions(initialMarkup, parameters))
+            {
+                var diagnostics = (await GetDiagnosticsAsync(workspace, parameters).ConfigureAwait(false)).ToImmutableArrayOrEmpty();
```

Comment:
sub-expression awaits are pretty annoying.  can we just assign this above, then operate on the result?